Makefile

```diff
@@ -125,14 +127,36 @@ build.k8srules:
 	@echo "--> building kubernetes routingrules controller"
 	@go build $(BUILDFLAGS) -ldflags '$(LDFLAGS)' -o $(BINDIR)/$(K8SRULES_APP_NAME) ./cmd/k8srules/
 
-build.cli: tools.go-bindata
-	@echo "--> building cli"
+build.cli.linux: tools.go-bindata
+	@echo "--> building cli for Linux"
+	@go-bindata -pkg=utils -prefix "./cli" -o ./cli/utils/i18n_resources.go ./cli/locales
+	@GOOS=linux GOARCH=amd64 go build $(BUILDFLAGS) -o $(BINDIR)/$(CLI_APP_NAME)-linux ./cmd/cli/
+	@goimports -w ./cli/utils/i18n_resources.go
+
+build.cli.darwin: tools.go-bindata
+	@echo "--> building cli for OS X"
+	@go-bindata -pkg=utils -prefix "./cli" -o ./cli/utils/i18n_resources.go ./cli/locales
+	@GOOS=darwin GOARCH=amd64 go build $(BUILDFLAGS) -o $(BINDIR)/$(CLI_APP_NAME)-darwin ./cmd/cli/
+	@goimports -w ./cli/utils/i18n_resources.go
+
+build.cli.windows: tools.go-bindata
+	@echo "--> building cli for Windows"
 	@go-bindata -pkg=utils -prefix "./cli" -o ./cli/utils/i18n_resources.go ./cli/locales
-	@$(foreach GOOS, $(TARGET_OS), env GOOS=$(GOOS) GOARCH=amd64 go build -ldflags '$(subst -linkmode external,,$(LDFLAGS))' -o $(BINDIR)/$(CLI_APP_NAME)-$(GOOS) ./cmd/cli/;) # Remove "-linkmode external" flag during build
-	@go build $(BUILDFLAGS) -ldflags '$(subst -linkmode external,,$(LDFLAGS))' -o $(BINDIR)/$(CLI_APP_NAME) ./cmd/cli/ # build an additional binary for the current OS
-	@mv $(BINDIR)/$(CLI_APP_NAME)-windows $(BINDIR)/$(CLI_APP_NAME)-windows.exe # add extension to windows binary
+	@GOOS=windows GOARCH=amd64 go build $(BUILDFLAGS) -o $(BINDIR)/$(CLI_APP_NAME)-windows.exe ./cmd/cli/
 	@goimports -w ./cli/utils/i18n_resources.go
 
+# build.cli: tools.go-bindata
```

Comment:
I kept this intentionally. The Makefile can use some more work in future, to coalesce targets. Didn't have the patience to do it all in one go. The commented code snippet would serve as a useful reference when we want to collapse the build targets in future.